app/controllers/internal_api/v1/profile_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class InternalApi::V1::ProfileController < InternalApi::V1::ApplicationController
+  def index
+    authorize :index, policy_class: ProfilePolicy
+    if current_user.avatar.attached?
+      avatar_url = url_for(current_user.avatar)
+    end
+    render json: { user: current_user.as_json.merge("avatar_url" => avatar_url) }, status: :ok
+  end
+
+  def remove_avatar
+    authorize :remove_avatar, policy_class: ProfilePolicy
```

Comment:
```suggestion
    authorize :profile
```

NIT: require same changes for update & index method

Reply:
@abinashpa can you please help me in resolving these comments?

Reply:
@onkar-saeloun These changes are to be done from my PR. You don't have to worry about it.